src/test/java/net/syscon/elite/repository/jpa/repository/CourtEventRepositoryTest.java

```diff
@@ -60,37 +64,51 @@ void setup() {
-                .offenderBooking(offenderBooking)
-                .offenderCourtCase(offenderBooking.getCourtCases().stream().findFirst().orElseThrow())
+                .offenderBooking(bookingWithCourtCase)
+                .offenderCourtCase(bookingWithCourtCase.getCourtCases().stream().findFirst().orElseThrow())
                 .startTime(startTime);
     }
 
     @Test
     void court_event_can_be_saved_and_retrieved_with_defaults_populated() {
-        final var savedCourtEvent = courtEventRepository.save(builder.build());
+        final var savedCourtEventWithCourtCase = courtEventRepository.save(builder.build());
 
         entityManager.flush();
 
-        assertThat(courtEventRepository.findById(savedCourtEvent.getId()).orElseThrow()).isEqualTo(savedCourtEvent);
+        assertThat(courtEventRepository.findById(savedCourtEventWithCourtCase.getId()).orElseThrow()).isEqualTo(savedCourtEventWithCourtCase);
 
         // defaults populated
-        assertThat(savedCourtEvent.getNextEventRequestFlag()).isEqualTo("N");
-        assertThat(savedCourtEvent.getOrderRequestedFlag()).isEqualTo("N");
+        assertThat(savedCourtEventWithCourtCase.getNextEventRequestFlag()).isEqualTo("N");
+        assertThat(savedCourtEventWithCourtCase.getOrderRequestedFlag()).isEqualTo("N");
     }
 
     @Test
     void court_event_can_be_saved_and_retrieved_with_defaults_overridden() {
-        final var savedCourtEvent = courtEventRepository.save(builder
+        final var savedCourtEventWithCourtCase = courtEventRepository.save(builder
                 .nextEventRequestFlag("X")
                 .orderRequestedFlag("Y")
                 .build());
 
         entityManager.flush();
 
-        assertThat(courtEventRepository.findById(savedCourtEvent.getId()).orElseThrow()).isEqualTo(savedCourtEvent);
+        assertThat(courtEventRepository.findById(savedCourtEventWithCourtCase.getId()).orElseThrow()).isEqualTo(savedCourtEventWithCourtCase);
 
         // defaults overridden
-        assertThat(savedCourtEvent.getNextEventRequestFlag()).isEqualTo("X");
-        assertThat(savedCourtEvent.getOrderRequestedFlag()).isEqualTo("Y");
+        assertThat(savedCourtEventWithCourtCase.getNextEventRequestFlag()).isEqualTo("X");
+        assertThat(savedCourtEventWithCourtCase.getOrderRequestedFlag()).isEqualTo("Y");
+    }
+
+    @Test
+    void court_event_without_court_case_retrieved() {
+        final var bookingWithoutCourtCase = offenderBookingRepository.findById(BOOKING_WITHOUT_COURT_CASE).orElseThrow();
+
+        assertThat(bookingWithoutCourtCase.getCourtCases()).isEmpty();
+
+        final var savedCourtEventWithoutCourtCase = courtEventRepository.save(builder
+                .offenderBooking(bookingWithoutCourtCase)
+                .offenderCourtCase(null)
+                .build());
+
+        assertThat(courtEventRepository.findById(savedCourtEventWithoutCourtCase.getId()).orElseThrow()).isEqualTo(savedCourtEventWithoutCourtCase);
```

Comment:
Meh.. could get rid of the orElseThrow() and do an `assertThat(...).contains()`

Could maybe also call the new method getOffenderCourtCase() to prove it returns empty?